components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -16,27 +17,157 @@ import DropDown from "../components/DropDown";
 import Modal from "../components/Modal";
 import Alert from "./Alert";
 
+interface hasId {
+    id: string;
+}
+
+type PendingStripeSubscription = { pendingSince: number };
+
 interface Props {
+    subject?: hasId;
     attributionId: AttributionId;
-    showSpinner: boolean;
-    showUpgradeBilling: boolean;
-    showManageBilling: boolean;
-    stripePortalUrl?: string;
-    usageLimit?: number;
-    doUpdateLimit: (newLimit: number) => Promise<void>;
 }
 
-export default function UsageBasedBillingConfig({
-    attributionId,
-    showSpinner,
-    showUpgradeBilling,
-    showManageBilling,
-    stripePortalUrl,
-    usageLimit,
-    doUpdateLimit,
-}: Props) {
+export default function UsageBasedBillingConfig({ subject, attributionId }: Props) {
+    const location = useLocation();
     const [showUpdateLimitModal, setShowUpdateLimitModal] = useState<boolean>(false);
     const [showBillingSetupModal, setShowBillingSetupModal] = useState<boolean>(false);
+    const [stripeSubscriptionId, setStripeSubscriptionId] = useState<string | undefined>();
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [stripePortalUrl, setStripePortalUrl] = useState<string | undefined>();
+    const [pollStripeSubscriptionTimeout, setPollStripeSubscriptionTimeout] = useState<NodeJS.Timeout | undefined>();
+    const [usageLimit, setUsageLimit] = useState<number | undefined>();
+    const [pendingStripeSubscription, setPendingStripeSubscription] = useState<PendingStripeSubscription | undefined>();
+    const [billingError, setBillingError] = useState<string | undefined>();
+
+    const localStorageKey = `pendingStripeSubscriptionFor${subject?.id || ""}`;
```

Comment:
Micro-nit: There is a super tiny ID collision risk if a user and a team both have the same ID. It's not super realistic, but using `AttributionId.render(attributionId)` instead of `subject?.id` would prevent that.

Reply:
Also, this code seems to use `AttributionId.render(attributionId)` (the string) more often than `attributionId` (the object). If that's true, maybe this code could be simplified a bit by refactoring `attributionId` to be a string and not a fully-hydrated shape. 💭 No strong opinion here though.

Reply:
I've applied both of these suggestions, thanks 👍